lifecycle_test.go

```diff
@@ -456,6 +461,166 @@ func deleteReleaseBundleWithProject(t *testing.T, lcManager *lifecycle.Lifecycle
 	time.Sleep(5 * time.Second)
 }
 
+func TestSetReleaseBundleTag(t *testing.T) {
+
+	cleanCallback := initLifecycleTest(t, artifactoryLifecycleSetTagMinVersion)
+	defer cleanCallback()
+	lcManager := getLcServiceManager(t)
+
+	deleteBuilds := uploadBuilds(t)
+	defer deleteBuilds()
+
+	//set tag
+	createRbFromSpec(t, tests.LifecycleBuilds12, tests.LcRbName1, number1, true, true)
+	defer deleteReleaseBundle(t, lcManager, tests.LcRbName1, number1)
+	assertStatusCompleted(t, lcManager, tests.LcRbName1, number1, "")
+	defer setReleaseBundleTag(t, lcManager, tests.LcRbName1, number1, "", "bundle-tag")
+
+	//unset tag
+	createRbFromSpec(t, tests.LifecycleBuilds12, tests.LcRbName2, number1, true, true)
+	defer deleteReleaseBundle(t, lcManager, tests.LcRbName2, number1)
+	assertStatusCompleted(t, lcManager, tests.LcRbName2, number1, "")
+	defer setReleaseBundleTag(t, lcManager, tests.LcRbName1, number1, "", "")
```

Comment:
similarly defer at end of the function is redundant, you can revisit other tests and utility functions